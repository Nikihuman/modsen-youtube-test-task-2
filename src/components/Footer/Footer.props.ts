import { HTMLAttributes } from 'react';

export interface FooterProps extends HTMLAttributes<HTMLElement> {}
