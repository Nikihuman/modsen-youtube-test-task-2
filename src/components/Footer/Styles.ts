import styled from 'styled-components';

export const StyledFooter = styled.footer`
  width: 100%;
  padding-top: 60px;
  border-top: 1px solid ${props => props.theme['border_dark']};
  margin-top: 50px;
`;

export const StyledLinksWrapper = styled.section`
  display: flex;
  @media ${props => props.theme['media']['phone']} {
    flex-direction: column;
    gap: 30px;
  }
`;

export const StyledLinksBlock = styled.div`
  width: 230px;
  display: flex;
  flex-wrap: wrap;
  column-gap: 7px;
  margin-right: 32px;
  @media ${props => props.theme['media']['extra_small']} {
    font-size: 16px;
  }
`;

export const StyledSocialLinksBlock = styled.div`
  display: flex;
  gap: 27px;
  margin-left: auto;
  @media ${props => props.theme['media']['phone']} {
    margin: 0 auto;
  }
`;

export const StyledCompanyName = styled.section`
  width: max-content;
  margin: 20px auto;
  font-size: 24px;
  font-weight: 500;
  color: ${props => props.theme['text_hover']};
  @media ${props => props.theme['media']['extra_small']} {
    font-size: 16px;
  }
`;
