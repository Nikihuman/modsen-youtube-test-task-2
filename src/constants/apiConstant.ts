export const SEARCH_BY_GENRE_PATH = (pageNum: number, genre: string, limit: number): string => {
  return `movie?page=${pageNum}&limit=${limit}&selectFields=id&selectFields=name&selectFields=alternativeName&selectFields=year&selectFields=poster&selectFields=genres&type=movie${genre !== 'все' ? `&genres.name=${encodeURI(genre)}` : ''}`;
};

export const SEARCH_BY_NAME_PATH = (pageNum: number, name: string, limit: number): string => {
  return `movie/search?page=${pageNum}&limit=${limit}&query=${encodeURI(name)}`;
};

export const SEARCH_BY_ID_PATH = (movieId: number): string => {
  return `movie/${movieId}`;
};

export const X_API_KEY = 'K2DWFBZ-ZDX45AZ-JBFZGKR-H82FQQR';
// export const X_API_KEY = 'BHHGG58-1VHMFRG-GNC0P54-16A61X6';  MY
export const BASE_URL = 'https://api.kinopoisk.dev/v1.4/';
