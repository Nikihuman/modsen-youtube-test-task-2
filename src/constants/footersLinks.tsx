import FacebookIcon from '@assets/images/facebook.svg';
import InstagramIcon from '@assets/images/instagram.svg';
import LinkedinIcon from '@assets/images/linkedin.svg';
import TwitterIcon from '@assets/images/twitter.svg';

export const FIRST_BLOCK = [
  {
    value: 'TermsPrivacyPolicy & Safety',
    link: '#',
  },
  {
    value: 'How YouTube works',
    link: '#',
  },
  {
    value: 'Test new features',
    link: '#',
  },
];

export const SECOND_BLOCK = [
  {
    value: 'About',
    link: '#',
  },
  {
    value: 'Press',
    link: '#',
  },
  {
    value: 'Copyright',
    link: '#',
  },
  {
    value: 'Contact us',
    link: '#',
  },
  {
    value: 'Creators',
    link: '#',
  },
  {
    value: 'Advertise',
    link: '#',
  },
  {
    value: 'Developers',
    link: '#',
  },
];

export const SOCIAL_MEDIA = [
  {
    value: <LinkedinIcon />,
    link: '#',
  },
  {
    value: <FacebookIcon />,
    link: '#',
  },
  {
    value: <InstagramIcon />,
    link: '#',
  },
  {
    value: <TwitterIcon />,
    link: '#',
  },
];
