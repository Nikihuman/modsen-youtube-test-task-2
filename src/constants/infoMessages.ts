export const INFO_MESSAGES = {
  video: 'Sorry, there are no trailers for this film in the database ðŸ˜¢',
  description: 'We`re sorry, we couldn`t find a descriptionðŸ˜¢',
  movieSearch: `We can't find any results by your requestðŸ˜¢`,
  namesSearch: 'Please enter the movie title (at least 2 characters)',
};
