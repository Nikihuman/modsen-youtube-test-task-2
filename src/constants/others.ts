export const COUNT_OF_GENRES_IN_MOVIE_CARD = 6;

export const PROFESSIONS = {
  director: {
    eng: 'director',
    rus: 'режиссер',
  },
};
