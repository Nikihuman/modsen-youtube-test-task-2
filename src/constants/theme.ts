export const THEME: ITheme = {
  primary: '#FF8A00',
  primaty_hover: '#c46c07',
  text_light: '#FFFFFF',
  text_dark: 'rgb(0, 0, 0)',
  text_hover: '#b7b5b5',
  dark: '#383838',
  medium: '#b7b7b7',
  border_dark: 'rgba(0, 0, 0, 0.1)',
  border_light: '#FFFFFF',
  light: '#f2f2f2',
  media: {
    big_tablet: '(max-width: 1440px)',
    tablet: '(max-width: 1080px)',
    phone: '(max-width: 720px)',
    big_tablet_card: '(min-width: 1275px) and (max-width: 1440px)',
    tablet_card: '(max-width: 1080px) and (min-width: 870px)',
    phone_card: '(max-width: 720px) and (min-width: 510px)',
    extra_small: '(max-width: 390px)',
  },
};

export interface ITheme {
  text_light: string;
  text_dark: string;
  text_hover: string;
  border_light: string;
  border_dark: string;
  primary: string;
  primaty_hover: string;
  dark: string;
  medium: string;
  light: string;
  media: {
    big_tablet: string;
    tablet: string;
    phone: string;
    big_tablet_card: string;
    tablet_card: string;
    phone_card: string;
    extra_small: string;
  };
}
