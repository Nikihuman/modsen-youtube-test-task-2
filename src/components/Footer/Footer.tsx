import { FIRST_BLOCK, SECOND_BLOCK, SOCIAL_MEDIA } from '@constants/footersLinks';
import {
  StyledCompanyName,
  StyledFooter,
  StyledLinksBlock,
  StyledLinksWrapper,
  StyledSocialLinksBlock,
} from './Styles';
import { Link } from '@components/Link/Link';
import { FooterProps } from './Footer.props';

export function Footer({ ...props }: FooterProps) {
  return (
    <StyledFooter {...props}>
      <StyledLinksWrapper>
        <StyledLinksBlock>
          {FIRST_BLOCK.map(el => (
            <Link key={el.value} href={el.link}>
              {el.value}
            </Link>
          ))}
        </StyledLinksBlock>
        <StyledLinksBlock>
          {SECOND_BLOCK.map(el => (
            <Link key={el.value} href={el.link}>
              {el.value}
            </Link>
          ))}
        </StyledLinksBlock>
        <StyledSocialLinksBlock>
          {SOCIAL_MEDIA.map((el, i) => (
            <Link key={i} href={el.link}>
              {el.value}
            </Link>
          ))}
        </StyledSocialLinksBlock>
      </StyledLinksWrapper>
      <StyledCompanyName>2023 Modsen company</StyledCompanyName>
    </StyledFooter>
  );
}
