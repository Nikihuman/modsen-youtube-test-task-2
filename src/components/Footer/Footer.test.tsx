import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { FIRST_BLOCK, SECOND_BLOCK, SOCIAL_MEDIA } from '../../constants/footersLinks';
import { THEME } from '../../constants/theme';
import { Footer } from './Footer';

test('Testing the Footer component', () => {
  render(
    <ThemeProvider theme={THEME}>
      <Footer data-testid="footer-element" />
    </ThemeProvider>,
  );
  const links = screen.getAllByRole('link');
  const footer = screen.getByTestId('footer-element');
  expect(footer).toBeInTheDocument();
  links.forEach(el => expect(el).toBeInTheDocument());
  expect(links.length).toEqual(FIRST_BLOCK.length + SECOND_BLOCK.length + SOCIAL_MEDIA.length);
  expect(footer).toMatchSnapshot('footer snapshot');
});
