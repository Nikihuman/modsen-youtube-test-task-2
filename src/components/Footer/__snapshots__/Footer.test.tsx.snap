// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Footer component: footer snapshot 1`] = `
<footer
  class="sc-guDLey lyUZs"
  data-testid="footer-element"
>
  <section
    class="sc-dmyCSP bZlFyv"
  >
    <div
      class="sc-hLQSwg iTwjgc"
    >
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        TermsPrivacyPolicy & Safety
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        How YouTube works
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        Test new features
      </a>
    </div>
    <div
      class="sc-hLQSwg iTwjgc"
    >
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        About
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        Press
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        Copyright
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        Contact us
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        Creators
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        Advertise
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        Developers
      </a>
    </div>
    <div
      class="sc-eDLKkx dhTFsk"
    >
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        <linkedin.svg />
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        <facebook.svg />
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        <instagram.svg />
      </a>
      <a
        class="sc-beySPh boPakf"
        href="#"
      >
        <twitter.svg />
      </a>
    </div>
  </section>
  <section
    class="sc-jTQCzO fIXjCO"
  >
    2023 Modsen company
  </section>
</footer>
`;
